engine-standalone-storage/src/native_ffi.rs

```diff
@@ -0,0 +1,89 @@
+use std::{
+    ffi::{self, OsStr},
+    ops::Deref,
+};
+
+use libloading::os::unix::{self, Library};
+use parking_lot::{Mutex, MutexGuard};
+use thiserror::Error;
+
+use aurora_engine::{contract_methods::ContractError, parameters::SubmitResult};
+
+static CONTRACT: Mutex<Option<DynamicContractImpl>> = Mutex::new(None);
+
+pub fn state() -> &'static super::state::State {
+    super::state::STATE.get().expect("state is not initialized")
+}
+
+#[inline]
+pub fn lock() -> impl Deref<Target = DynamicContractImpl> {
+    MutexGuard::map(CONTRACT.lock(), |x| {
+        x.as_mut().expect("must load library before use")
+    })
+}
+
+#[allow(dead_code)]
+pub struct DynamicContractImpl {
+    library: Library,
+    get_version_fn: extern "C" fn() -> *mut ffi::c_void,
+    submit_fn: extern "C" fn() -> *mut ffi::c_void,
+    submit_with_args_fn: extern "C" fn() -> *mut ffi::c_void,
```

Comment:
I guess there should be a map like: `HashMap<&'static str, extern "C" fn() -> *mut ffi::c_void>`?